pkg/controller/controller_ref_manager.go

```diff
@@ -107,6 +107,12 @@ func (m *BaseControllerRefManager) ClaimObject(ctx context.Context, obj metav1.O
 		// Ignore if the object is being deleted
 		return false, nil
 	}
+
+	if m.Controller.GetNamespace() != obj.GetNamespace() {
```

Comment:
only apply this check if the controller is actually namespaced:
```suggestion
	if len(m.Controller.GetNamespace()) > 0 && m.Controller.GetNamespace() != obj.GetNamespace() {
```